src/Compilers/CSharp/Portable/Binder/Semantics/OverloadResolution/OverloadResolution.cs

```diff
@@ -3349,6 +3349,13 @@ internal EffectiveParameters(ImmutableArray<TypeWithAnnotations> types, Immutabl
                 isMethodGroupConversion,
                 allowRefOmittedArguments);
 
+            // If we were producing complete results and had missing arguments, we pushed on in order to call IsApplicable for
+            // type inference and lambda binding. In that case we still need to return the argument mismatch failure here.
+            if (completeResults && !argumentAnalysis.IsValid)
+            {
+                return new MemberResolutionResult<TMember>(member, leastOverriddenMember, MemberAnalysisResult.ArgumentParameterMismatch(argumentAnalysis), hasTypeArgumentInferredFromFunctionType: false);
+            }
```

Comment:
I'm assuming `IsApplicable` can't have an effect on the returned result so it's safe to move this before `IsApplicable`. But draft for now to see what CI has to say.

Reply:
Ok there are lots of failling tests.

It seems like `IsApplicable` has a side effect somehow.

I'm thinking there could still be a way to reduce the work done by `IsApplicable` in this code path, but not sure. This needs more debugging to understand.